shared/instancewriter/instance_tar_writer.go

```diff
@@ -100,10 +100,21 @@ func (ctw *InstanceTarWriter) WriteFile(name string, srcPath string, fi os.FileI
 
 	// Handle xattrs (for real files only).
 	if link == "" {
-		hdr.Xattrs, err = shared.GetAllXattr(srcPath)
+		xattrs, err := shared.GetAllXattr(srcPath)
 		if err != nil {
 			return errors.Wrapf(err, "Failed to read xattr for %q", srcPath)
 		}
+
+		hdr.PAXRecords = make(map[string]string, len(xattrs))
+		for key, val := range xattrs {
+			if key == "system.posix_acl_access" {
+				hdr.PAXRecords["SCHILY.acl.access"] = val
+			} else if key == "system.posix_acl_default" {
+				hdr.PAXRecords["SCHILY.acl.default"] = val
+			} else {
+				hdr.PAXRecords["SCHILY.xattr."+key] = val
+			}
+		}
```

Comment:
So one thing I'm confused about is why don't we need to shift/unshift those records like we do the uid/gid above (see 78)?

I would have expected the ACL to be similarly shifted (as we have shifting logic in idmap for it) and so would have expected having to load up the ACLs, unshift the records and write the unshifted into the tarball.

If this logic also covers the fscaps, then I would similarly expect us to need to pull the V3 fscaps, get the rootid from it and unshift that if the container is running with a filesystem idmap set.